File: Axiom/Internal/Ast/ForStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Axiom.Internal.Ast
{
    internal class ForStatement : Statement
    {
        public ForStatement(Statement init, Expression condition, Statement loop, Statement body)
        {
            Init = init;
            Condition = condition;
            Loop = loop;
            Body = body;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Statement Init { get; private set; }
        public Expression Condition { get; private set; }
        public Statement Loop { get; set; }
        public Statement Body { get; set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/ForStatement.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;

public class ForStatement extends Statement {

	private Statement init;
	private Expression condition;
	private Statement loop;
	private Statement body;

	public ForStatement(Position position, Statement init, Expression condition, Statement loop, Statement body) {
		super(position);
		this.init = init;
		this.condition = condition;
		this.loop = loop;
		this.body = body;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Statement getInit() {
		return init;
	}

	public Expression getCondition() {
		return condition;
	}

	public Statement getLoop() {
		return loop;
	}

	public Statement getBody() {
		return body;
	}

}
